andela/caret-bn-backend
# #168781714 add chat functionality

#### What does this PR do?
Add chat functionality

#### Description of Task to be completed?
- Adds chat functionality for logged in users
- Adds the chat interface

#### How should this be manually tested?
- Clone the repository from [here](https://github.com/andela/caret-bn-backend.git)
- Browse to the repo directory
- navigate to ft-users-chat-168781714 branch
- Run `npm install`
- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`

To chat

- In Postman, login and copy the token
- Paste the token into the local storage of the browser
- In the browser, navigate to localhost:3000
You should see an interface that provides the text input and send button
Type your messages, hit send, and your messages as well as those from everyone connected should appear in the console

To view the chat history

- In Postman Provide a login bearer token
- send a `GET` request to `http://localhost:<:port>/api/v1/chat`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#168781714
#### Screenshots (if appropriate)
<img width="1264" alt="Screen Shot 2019-11-06 at 18 06 15" src="https://user-images.githubusercontent.com/50104386/68315381-5bace600-00c0-11ea-8440-a9ccd61f2a00.png">
<img width="1440" alt="Screen Shot 2019-11-06 at 18 06 33" src="https://user-images.githubusercontent.com/50104386/68315383-5c457c80-00c0-11ea-87cb-3460459be0ad.png">
<img width="1173" alt="Screen Shot 2019-11-06 at 18 07 35" src="https://user-images.githubusercontent.com/50104386/68315384-5c457c80-00c0-11ea-8be8-64923aac8c77.png">

#### Questions:
N/A
